packages/ddp-server/livedata_server.js

```diff
@@ -1718,8 +1718,19 @@ DDPServer._calculateVersion = calculateVersion;
 // "blind" exceptions other than those that were deliberately thrown to signal
 // errors to the client
 var wrapInternalException = function (exception, context) {
-  if (!exception || exception instanceof Meteor.Error)
+  if (!exception) return exception;
+
+  // To allow packages to throw errors intended for the client but not have to
+  // depend on the Meteor.Error class, `isClientSafe` can be set to true on any
+  // error before it is thrown.
+  if (exception.isClientSafe) {
+    if (!(exception instanceof Meteor.Error)) {
+      const originalMessage = exception.message;
+      exception = new Meteor.Error(exception.error, exception.reason, exception.details);
```

Comment:
Are we sure that these properties (`exception.error, exception.reason, exception.details`) will be defined (or safely undefined) for arbitrary `exception` objects that happen to use the `exception.isClientSafe` convention? Should `Meteor.Error` take in the whole `exception` object and copy its properties (e.g. `.stack`)?